File: app/src/main/java/com/pietrantuono/podcasts/singlepodcast/viewmodel/PodcastEpisodeViewModel.java
package com.pietrantuono.podcasts.singlepodcast.viewmodel;

import android.graphics.drawable.Drawable;
import android.support.annotation.Nullable;

import com.pietrantuono.Constants;
import com.pietrantuono.podcasts.R;
import com.pietrantuono.interfaceadapters.apis.PodcastEpisodeModel;

public class PodcastEpisodeViewModel extends BasePodcastEpisodeViewModel {
    private final ResourcesProvider resourcesProvider;

    public PodcastEpisodeViewModel(PodcastEpisodeModel podcastEpisodeModel, ResourcesProvider resourcesProvider) {
        super(podcastEpisodeModel);
        this.resourcesProvider = resourcesProvider;
    }

    public VisibilityWrapper getVisibilityOfFooter() {
        if (getVisibilityOfType().isVisible() || getVisibilityOfDuration().isVisible()) {
            return VisibilityWrapper.VISIBLE;
        } else {
            return VisibilityWrapper.GONE;
        }
    }

    public VisibilityWrapper getVisibilityOfType() {
        if (getMediaTypeText() == null) {
            return VisibilityWrapper.GONE;
        } else {
            return VisibilityWrapper.VISIBLE;
        }
    }

    public VisibilityWrapper getVisibilityOfDuration() {
        if (getDuration() == null || getDuration().isEmpty()) {
            return VisibilityWrapper.GONE;
        } else {
            return VisibilityWrapper.VISIBLE;
        }
    }

    @Nullable
    public Drawable getMediaTypeImage() {
        try {
            String type = getEnclosures().get(0).getType().toLowerCase();
            return getImageResouce(type);
        } catch (NullPointerException | IndexOutOfBoundsException e) {
            return null;
        }
    }

    @Nullable
    private Drawable getImageResouce(String type) {
        if (type.contains(Constants.AUDIO)) {
            return resourcesProvider.ContextCompatgetDrawable(R.drawable.ic_audio_icon);
        }
        if (type.contains(Constants.VIDEO)) {
            resourcesProvider.ContextCompatgetDrawable(R.drawable.ic_video_icon);
        }
        return null;
    }

    @Nullable
    public String getMediaTypeText() {
        try {
            String type = getEnclosures().get(0).getType().toLowerCase();
            return getStringResource(type);
        } catch (NullPointerException | IndexOutOfBoundsException e) {
            return null;
        }
    }

    @Nullable
    private String getStringResource(String type) {
        if (type.contains(Constants.AUDIO)) {
            return resourcesProvider.getString(R.string.audio);
        }
        if (type.contains(Constants.VIDEO)) {
            return resourcesProvider.getString(R.string.video);
        }
        return null;
    }

}

File: app/src/main/java/com/pietrantuono/podcasts/singlepodcast/viewmodel/PodcastEpisodeViewModel.kt
package com.pietrantuono.podcasts.singlepodcast.viewmodel

import android.graphics.drawable.Drawable

import com.pietrantuono.podcasts.Constants
import com.pietrantuono.podcasts.R
import com.pietrantuono.podcasts.apis.PodcastEpisodeModel

class PodcastEpisodeViewModel(podcastEpisodeModel: PodcastEpisodeModel, private val resourcesProvider:
ResourcesProvider) : BasePodcastEpisodeViewModel(podcastEpisodeModel) {

    val visibilityOfFooter: VisibilityWrapper
        get() =
        if (visibilityOfType.isVisible || visibilityOfDuration.isVisible) {
            VisibilityWrapper.VISIBLE
        } else {
            VisibilityWrapper.GONE
        }

    val visibilityOfType: VisibilityWrapper
        get() =
        if (mediaTypeText == null) {
            VisibilityWrapper.GONE
        } else {
            VisibilityWrapper.VISIBLE
        }

    val visibilityOfDuration: VisibilityWrapper
        get() =
        if (duration == null || duration!!.isEmpty()) {
            VisibilityWrapper.GONE
        } else {
            VisibilityWrapper.VISIBLE
        }

    val mediaTypeImage: Drawable?
        get() =
        try {
            val type = enclosures!![0].type.toLowerCase()
            getImageResouce(type)
        } catch (e: Exception) {
            null
        }

    private fun getImageResouce(type: String): Drawable? {
        if (type.contains(Constants.AUDIO)) {
            return resourcesProvider.ContextCompatgetDrawable(R.drawable.ic_audio_icon)
        }
        if (type.contains(Constants.VIDEO)) {
            resourcesProvider.ContextCompatgetDrawable(R.drawable.ic_video_icon)
        }
        return null
    }

    val mediaTypeText: String?
        get() =
        try {
            val type = enclosures!![0].type.toLowerCase()
            getStringResource(type)
        } catch (e: Exception) {
            null
        }


    private fun getStringResource(type: String): String? {
        if (type.contains(Constants.AUDIO)) {
            return resourcesProvider.getString(R.string.audio)
        }
        if (type.contains(Constants.VIDEO)) {
            return resourcesProvider.getString(R.string.video)
        }
        return null
    }

}
